x-pack/plugins/fleet/server/services/epm/packages/get.ts

```diff
@@ -163,10 +163,16 @@ export async function getPackageFromSource(options: {
     // else package is not installed or installed and missing from cache and storage and installed from registry
     res = await Registry.getRegistryPackage(pkgName, pkgVersion);
   }
+
   if (!res) throw new Error(`package info for ${pkgName}-${pkgVersion} does not exist`);
+
+  const { paths, packageInfo } = res;
+  setArchiveFilelist({ name: pkgName, version: pkgVersion }, paths);
```

Comment:
Not sure if we want to set every time or only if we think it changed, bit at least this will cache results for later